direct_connect.go

```diff
@@ -0,0 +1,187 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/netip"
+	"sync"
+	"time"
+
+	"io"
+	"log"
+	"net/http"
+
+	"golang.zx2c4.com/wireguard/conn"
+	"golang.zx2c4.com/wireguard/device"
+	"golang.zx2c4.com/wireguard/tun/netstack"
+	gtcp "gvisor.dev/gvisor/pkg/tcpip/transport/tcp"
+	gudp "gvisor.dev/gvisor/pkg/tcpip/transport/udp"
+
+	"golang.zx2c4.com/wireguard/wgctrl/wgtypes"
+
+	"wiretap/peer"
+	"wiretap/transport/icmp"
+	"wiretap/transport/tcp"
+	"wiretap/transport/udp"
+
+	"github.com/rainforestapp/rainforest-cli/rainforest"
+	"github.com/urfave/cli"
+)
+
+func launchDirectConnect(c cliContext, rfApi *rainforest.Client) error {
```

Comment:
much of this function is a trimmed down version of what's in wiretap's serve cmd.